services/management/actions.go

```diff
@@ -0,0 +1,217 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+var (
+	errUnsupportedAction  = errors.New("unsupported action")
+	errPmmAgentIDNotFound = errors.New("can't detect pmm_agent_id")
+)
+
+type agentsRegistry interface {
+	// SendRequest sends request to pmm-agent with given id.
+	SendRequest(ctx context.Context, pmmAgentID string, payload agentpb.ServerRequestPayload) agentpb.AgentResponsePayload
+}
+
+// ActionsService describes an Actions Application Service.
+// Provides functions for PMM Actions manipulation.
+type ActionsService struct {
+	agentsRegistry agentsRegistry
+	actionsStorage *InMemoryActionsStorage
```

Comment:
That's already `ActionsService`. No need to be that explicit. `registry registry`, `storage *InMemoryActionsStorage`.